src/python/bot/fuzzers/engine.py

```diff
@@ -47,6 +49,10 @@ def __init__(self, crashes, stats):
 class Engine(object):
   """Base interface for a grey box fuzzing engine."""
 
+  @property
+  def name(self):
+    return self.__class__.__name__
```

Comment:
This will need to be overridden for AFL which we call "afl" everywhere but in class names.
I also think we may want to do the same for libFuzzer which we generally call "libFuzzer" and not "LibFuzzer". In general I don't like the idea of python class names (which obey certain conventions) determining the name we use elsewhere.
I'm not sure this property is a good idea due to this problem, it seems like it can introduce subtle problems.

Reply:
Replaced this with `raise NotImplementedError` so that we explicitly specify the name.